src/mame/konami/windy2.h

```diff
@@ -0,0 +1,67 @@
+// license:BSD-3-Clause
+// copyright-holders:windyfairy
+/*
+ * Konami Windy2 I/O (JVS)
+ *
+ */
+#ifndef MAME_MACHINE_WINDY2_H
+#define MAME_MACHINE_WINDY2_H
+
+#pragma once
+
+#include "machine/jvsdev.h"
+#include "machine/timer.h"
+
+class windy2_device : public jvs_device
+{
+public:
+	enum : uint8_t {
+		// Corresponds to positions on DIPSW8 that controls panel type
+		PANEL_2L6B  = 0b00000010, // "Windy II Standard" 2L6B Panel (2x 1L3B)
+		//PANEL_MAHJONG  = 0b11111111, // Mahjong Panel (23B) (GU707-JA), dipswitch setting unknown
+		PANEL_QUIZ  = 0b00000000, // Quiz Panel (4B) (GU707-JB)
+		PANEL_2L8B  = 0b00000011, // 2L8B Panel (2x 1L4B) (GU707-JC)
+		PANEL_1L6B  = 0b10010101, // 1L6B Panel (GU707-JD)
+		PANEL_2L12B = 0b00010101, // 2L12B Panel (2x 1L6B) (GU707-JG)
+	};
+
+	template <typename T>
+	windy2_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock, T &&jvs_host_tag, uint8_t panel_type)
+		: windy2_device(mconfig, tag, owner, clock)
+	{
+		host.set_tag(std::forward<T>(jvs_host_tag));
+		m_panel_type = panel_type;
+	}
+
+	windy2_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	virtual ioport_constructor device_input_ports() const override;
+
+	DECLARE_INPUT_CHANGED_MEMBER(coin_inserted);
+
+protected:
+	template <uint8_t First> void set_port_tags() { }
```

Comment:
What’s this for?  It doesn’t appear to be implemented.

Reply:
I copied the initial template I've been using for JVS devices from one of the older Sega JVS device implementations and for some reason didn't remove it. I think it's been in all of the JVS devices I've implemented so far so I'll remove it from all of them.